### DISCUSSION TITLE
Cannot get web interface to work

### DISCUSSION THREAD

**Discussion by AlfredPerson** [Motion-Project/motionplus#149] [General]
Having just upgraded from motion to motionplus, I seem to be having a few issues.
In motion, I had multiple cameras working and found the web interface quite useful.
There are a number of features of motionplus that I wanted to use, so installed motionplus.
I modified my .conf files (I use a motionplus.conf and one .conf file per camera) to reflect (hopefully) the changes needed.
motionplus is working and capturing videos on all cameras, but there is no audio stream recorded on any captured video (all cameras have built in microphones).
I have set the webcontrol_localhost to off so that I can view the site remotely - but this doesn't seem to work despite being the same host address as when using motion, so I know the address/ port is accessible.
When I view the page using localhost:[port], I don't see any streams and it doesn't appear as it did in motion - when I go to the config page, I cannot configure any camera - as no camera appears in the pull-down box for the cameras.
I suspect there is a parameter I am not setting correctly in the .conf file, even though I have compared my file against the sample one provided.
Therefore, I think I have two questions - how can I get audio added to the video streams, and how may I get the streams visible in the webcontrol html?
(I also looked for the streams using [ipaddress]:[port]/[cameraif]mjpg without success.)
Many thanks for your help.

**Comment by Mr-Dave** [comment]
You should start by reviewing the log to determine whether the webcontrol started as well as seeing if it reported any unknown parameters.  (Log level INF or greater).  Next, you'd want to make sure that review the port  being specified.  In Motionplus, only one port is used as shown in the [examples](https://motion-project.github.io/motionplus_examples.html#webcontrol_pages) whereas in Motion there were multiple ports per camera.

Finally, just to set expectations.  The web interface does not provide sound from the cameras.  I asked for insights from the community and didn't get any responses.  (Sound is attempted on passthrough videos only)

**Comment by AlfredPerson** [comment]
Many thanks for the response.

I have logging set, and prior to that,  ensured there were no issues with incorrect parameters (as I wanted to minimise the log file!).

The information in the log file does not seem to have any errors, but does have each trigger and subsequent generation of video.

I have set webcontrol_localhost to be off, so I would expect to be able to view the result across the network, but cannot.

I only have one port defined for the webcontrol - and therefore, I know that the IP address and port work across the network.

I could not see any issues with the webcontrol starting in the log file - or on the command line, so I am at a loss as to the issue - so any help would be appreciated.

When accessing the webcontrol site, I get the same display I had in motion, i.e. the green/ black background, with the "menu" 3 bars in the top left, but no streams; if I select the configuration page, I can see lots of options, but no cameras are visible in the selection box.

As for the audio, I know the cameras are generating audio, because if I use e.g. VLC and access the camera stream as defined in the camer .conf files, I can hear the audio.

When I look at the video files saved by motionplus and use e.g. VLC and look at the codec information, there is information about the video, but none about audio.

I just thought - in motion, I used the .mkv format, is that why there is no audio?

As for adding audio to the web display, I think that would be nice, especially if each camera's audio could be enabled/ disabled easily - my argument for that is that assuming one uses the web display of feeds to monitor what is going on, and there is an event occurring on a stream, it would be useful to enable audio on that stream in case it is something that needs action (e.g. an animal playing or is it in distress).

**Comment by bitraid** [comment]
For audio capture, make sure you have `movie_passthrough on` in each camera configuration file.
For remotely accessing the web interface, make sure that the port is being forwarded (by using, for example, https://portchecker.co/).
As for the web interface not displaying any streams (while there are no log errors and the capture works), maybe you have some browser extension that blocks javascript? have you tried a different browser?

**Comment by AlfredPerson** [comment]
@bitraid - many thanks for your help.
I am confident that the port and web address work - as I am using exactly the same port and address as I did for motion, which worked.
To clarify, using localhost:[port] and I get a web page - but no access to any streams/ cameras/ ...
Using [ip_address]:[port] from a different device no longer works, but it used to with motion.
I have double checked the firewall, but maybe tomorrow I will try a traceroute or similar.
I tried the movie_passthrough - and I now seem to have a few incredibly short videos, but I will investigate more - but I suspect maybe detection is now on audio AND video - whereas I only need video motion detection. 
I did check the videos, and now see that there is audio included in the captured file, so I think that works, thank you.

**Comment by AlfredPerson** [comment]
My apologies, but I am still at a loss.
The locolhost:[port] still only shows a blank green screen, so I cannot configure/ or see any of the streams.
I also tried accessing the camera streams on the local host and was unable to see them (http://[ip_address]:[port]/[camera_id]/mjpg)
The remote access still doesn't work at all - but that is secondary as the localhost isn't functional.
I am still capturing videos, but need to review later today to check for audio.
Any more clues would be most appreciated.

**Comment by Mr-Dave** [comment]
You'd need to post in a gist or somewhere the complete Motionplus log at log_level 8.  Also probably make sure that you do not have multiple instances of Motion and Motionplus running at the same time or using a configuration file different than what you expected.

**Reply by AlfredPerson** [reply]
I'll add the log, but many lines are identical I suspect.
I have checked that I only have a single instance of motionplus running (I do recall having issues with multiple instances of motion originally, before migrating to motionplus).
I have checked and am unaware of any issues with the motionplus.conf or camera.conf files.
I do appreciate the help!

**Reply by AlfredPerson** [reply]
Is there a part of the log file that may be of interest, as it is (currently) over 5MB long.
I can confirm that the config file is being identified for the cameras and motionplus.
I have noticed a number of aspects that puzzling to me, including that I have defined text to appear in the videos, and the format is correct, but the text (text_left & text_right), but no text is displayed on any videos that I checked.
A common information level message is regarding the despeckle filter, which it reports " mlp_setupmode:changes after 'EedDl':     0 - labels:   0 - noise level: 21"
If I could eliminate that notification, the log file would be a fraction of its current length!
A common feature of the videos is that they seem incredibly short (much shorter than I would have in motion), and often there seems to be maybe seconds of a "video" at the beginning without any video.
Many thanks for your help.

**Comment by Prhewett** [comment]
I have a similar problem as AlfredPerson with the web interface - I am unable to make any configuration changes. 
It's a fresh install of Raspbian Bookworm on a raspberry Pi 3 B and a Pi camera V2 .
I did a fresh install of moptionplus from the deb package. 
I configured logging level 8  and it simply says no camera config files found - awiaitng config in web intereface - but it doens't allow me to change the config in the web interface....
Log file attached. Also a link to a video clip of the behaviour I am seeing in the web interface . note it is the same on localhost:8080 as on a separate device on the LAN @  192.168.86.44:808. 
[motionplus.log](https://github.com/user-attachments/files/16438453/motionplus.log)
https://drive.google.com/drive/folders/1bGGc7dB6H-hApjjXN9ioIWsoX-r7tt3P?usp=sharing

**Reply by mollmann-jim** [reply]
Do you have a "camera" defined in your config file? I have:

camera /home/xyz/.motionplus/PiCamera3Wide.conf

Your config file is:

Jul 31 18:12:48 [INF][ALL][00:motionplus] Config file: /etc/motionplus/motionplus.conf

**Reply by Prhewett** [reply]
yes @mollmann-jim .  I have camera defined in motionplus.conf as:  /etc/motionplus/camera1.conf
here are the full file contains:
`; /usr/etc/motionplus/camera1.conf
;
; This config file was generated by motionplus 0.2.1

;*************************************************
; Configuration options specific to camera 1
;*************************************************

;*************************************************
;*****   Camera
;*************************************************
device_name RaspberryPi4
device_id 101

;*************************************************
;*****   Source
;*************************************************
;netcam_url rtsp://yourcamera1ip:port/camera/specific/url
libcam_device camera0
;*************************************************
;*****   Image
;*************************************************
width 1280
height 720

;*************************************************
;*****   Overlays
;*************************************************
text_left Pi3

;*************************************************
;*****   Movie
;*************************************************
movie_filename Pi4_%t-%v-%Y%m%d%H%M%S

**Reply by Mr-Dave** [reply]
It does not appear that that file has been specified in the motionplus.conf file.  (The log file you posted shows that it is not recognized.).  Additionally, for the parameters being locked via the web control, this is controlled via the [webcontrol_parms](https://motion-project.github.io/motionplus_config.html#webcontrol_parms)

**Reply by Prhewett** [reply]
ahhh - i didnt realise i had to change the defaults. Yep - that got it working - Thanks !

**Reply by Prhewett** [reply]
I have a few other web interface issues - but I'll turn up the logging again and see if I can work them out for myself. Thanks again @Mr-Dave

**Comment by Anonymouse83** [comment]
Sadly, I ended up having to uninstall MotionPlus and reinstalling Motion - the problems were getting worse over days to the point, where I think it corrupted the OS, so I needed to reinstall the OS as well.
I would still love to have MotionPlus working with audio, but would need far more time than I have to do so.
Sorry!

**Reply by Prhewett** [reply]
audio is next for me - the video seem to work pretty well really ! And I've tinkered with the motion sening a little to get it to stop triggering on tree movements. works well !
